packages/grafana-ui/src/components/Logs/logParser.ts

```diff
@@ -14,16 +14,24 @@ type FieldDef = {
 };
 
 /**
- * Returns all fields for log row which consists of fields we parse from the message itself and any derived fields
- * setup in data source config.
+ * Returns all fields for log row which consists of fields we parse from the message itself and additional fields
+ * found in the dataframe (they may contain links).
  */
 export const getAllFields = memoizeOne(
   (row: LogRowModel, getFieldLinks?: (field: Field, rowIndex: number) => Array<LinkModel<Field>>) => {
-    const fields = parseMessage(row.entry);
-    const derivedFields = getDerivedFields(row, getFieldLinks);
-    const fieldsMap = [...derivedFields, ...fields].reduce((acc, field) => {
+    const logMessageFields = parseMessage(row.entry);
+    const dataframeFields = getDataframeFields(row, getFieldLinks);
+    const fieldsMap = [...dataframeFields, ...logMessageFields].reduce((acc, field) => {
       // Strip enclosing quotes for hashing. When values are parsed from log line the quotes are kept, but if same
       // value is in the dataFrame it will be without the quotes. We treat them here as the same value.
+      // we need to handle this scenario:
```

Comment:
One small nit from my part: I would respect the capitalization used in the rest of the file. Absolutely minor, but looks better IMO.

Reply:
good point, updated.